graphql/util.lua

```diff
@@ -38,6 +47,20 @@ function util.coerceValue(node, schemaType, variables)
     return variables[node.name.value]
   end
 
+  if schemaType.__type == 'Union' then
+    local matched, result
+    for k, v in ipairs(schemaType.types) do
+        matched, result = pcall(util.coerceValue, node, v, variables)
+        if matched then
+          return result
+        end
+    end
+
+    local valid_types = util.map(schemaType.types, function (t) return t.name end)
+    --TODO! how to better report this error?
+    error('Could not coerce "' .. tostring(node.kind) .. '" to one of union types ['..table.concat(valid_types, ",")..']')
+  end
```

Comment:
There are discussions about this here https://github.com/facebook/graphql/issues/114 and here https://github.com/graphql/graphql-js/issues/207
I guess i'll remove the code about unions for now (but there are still problems in handling unions, resolveType function is not enforced and there is no default resolver for that)